### PR TITLE
MinimumViableSpacing on VariableDeclarations

### PR DESCRIPTION
**Pull request by aegershman**
closes https://github.com/openrewrite/rewrite/issues/268
Noticed this while working on https://github.com/openrewrite/rewrite/issues/197

This places at least one minimal spaces between `J.Modifier`'s and `TypeExpressions` on a variable declaration, as well as between the `typeExpression` and the variable identifier(s).

### CODE REVIEW

File: rewrite-test/src/main/kotlin/org/openrewrite/java/format/MinimumViableSpacingTest.kt [hunk 1]
```diff
@@ -59,4 +59,62 @@ interface MinimumViableSpacingTest : RecipeTest {
...
+            class A {
+                int unassigned;
+                int zero = 0;
+                final int one = 1;
+                public static final int ONE = 1;
+                public static final int TWO = 1, THREE = 3;
+            }
+        """,
+        after = """
+            class A {int unassigned;int zero=0;final int one=1;public static final int ONE=1;public static final int TWO=1,THREE=3;}
```
Comment (jkschneider) [inline rewrite-test/src/main/kotlin/org/openrewrite/java/format/MinimumViableSpacingTest.kt:76]: @aegershman the intent of `MinimumViableSpacing` is to ensure there is at least some space, not to strip out spacing that already exists. Consider the method declaration `public void method()`. If we strip out the spacing between the modifier and return type, we're left with an uncompilable form `publicvoid method()`. So in this case, minimum viable spacing ensures that there is at least _some_ space between modifiers and return type.

**Review by jkschneider — CHANGES_REQUESTED** [review]

File: rewrite-test/src/main/kotlin/org/openrewrite/java/format/MinimumViableSpacingTest.kt (hunk 1 above)
Comment (aegershman) [reply rewrite-test/src/main/kotlin/org/openrewrite/java/format/MinimumViableSpacingTest.kt:76]: @jkschneider right that's what this is trying to solve because it's **currently** broken like that where we'd get `publicvoid method()` or `finalstaticintONE=1`-- this is only adding spacing if it doesn't already exist. I'm under the impression the reason it looks like we're removing all spacing is because that's part of "proving" the minimum viable spacing is in action. It's this section here: https://github.com/openrewrite/rewrite/blob/0ea0a08a19604d6854faf5b16a5dff1ee7e37c99/rewrite-test/src/main/kotlin/org/openrewrite/java/format/MinimumViableSpacingTest.kt#L27-L32

This test works by first stripping out all spacing and then passing it through `MinimumViableSpacing` in order to demonstrate the spacing works at all, as shown in the test previous here: https://github.com/openrewrite/rewrite/blob/0ea0a08a19604d6854faf5b16a5dff1ee7e37c99/rewrite-test/src/main/kotlin/org/openrewrite/java/format/MinimumViableSpacingTest.kt#L37-L45. 

TL;DR if you comment out the changes made in `MinimumViableSpacingVisitor`, we get lines like `finalintone=1`, and `publicstaticfinalintONE=1` which is what this was trying to fix to begin with 🤷

**Review by jkschneider — APPROVED** [review]
